### ISSUE TITLE
XM:i SAM Optional field ... seems like hits + 1 for all hits?

### ISSUE THREAD

**Issue by MikeAxtell** [BenLangmead/bowtie#127]
I'm having a hard time understanding the XM:i optional SAM tag. I observe that for unmapped reads, it is set to 0, which I expect based on the documentation. But I also observe that for reads with one or more valid hits, but below the `-m` cutoff and therefore not suppressed, the number shown by XM:i always exceeds the actual number of hits by 1.

The documentation only discusses the XM:i tag for unmapped reads, or those that are suppressed by -m. But what about for 'normal' hits that do get output? Always the number of hits + 1?

Example call (version 1.3.0)

`bowtie -p 3 -v 1 -m 50 -S -a --best --strata -x ref.fa reads.fq.gz > out.sam`

As an example, here's a set of SAM lines for a read that had exactly three reported hits in the SAM file .. XM:i is set to 4.

```
SRR3222444.62	0	2	9353	255	12M	*	0	0	GTCCGTTCGGCG	DDDDDIIIIIII	XA:i:0	MD:Z:12	NM:i:0	XM:i:4
SRR3222444.62	0	3	14192915	255	12M	*	0	0GTCCGTTCGGCG	DDDDDIIIIIII	XA:i:0	MD:Z:12	NM:i:0	XM:i:4
SRR3222444.62	0	3	14203324	255	12M	*	0	0GTCCGTTCGGCG	DDDDDIIIIIII	XA:i:0	MD:Z:12	NM:i:0	XM:i:4
```

Thanks!

**Comment by ch4rr0** [comment]
This seems like a bug to me. I pushed a change to the `bug_fixes` branch that should resolve this.

**Comment by MikeAxtell** [comment]
Nice. Confirmed that commit solved the issue; XM:i now agrees with actual number of alignments reported.

Thanks!!

**Comment by ch4rr0** [comment]
This change is now available in our latest release, v1.3.1.
